### DISCUSSION TITLE
ESP32 on NodeMCU with Potentiometer - How to send MQTT Messages on change?

### DISCUSSION THREAD

**Discussion by DeepHyperspace** [arendst/Tasmota#17748] [Q&A]
Hey there,

i have a ESP32 on NodeMCU with Potentiometer connected on GPIO34. 
In the configuration i have set `IA GPIO34 to ADC Range 1`.

So it says on the main page: `Analog1 Range | 66` so it shows values between 0 and 100, good.

Currently it sends only MQTT Messages in the Interval set with the Tele-Period setting. For example on default every 300 seconds.
I want to change that it is event driven. So when the value changes it should immediately send an new MQTT Message. As I understand it, you have to define a rule for this.

There is an example here: https://tasmota.github.io/docs/Rules/#use-a-potentiometer

So i tried this: `Rule ON analog#a0div10 DO publish cmnd/grouplight/dimmer %value% ENDON`

But i cannot see any new MQT Messages in the Consoles sent, when i turn the Potentiometer. So it seems like the rule is not working.

The rule is active using Rule1 1 ;-)
I also tried analog#a1 insted of analog#a0.

So where is my mistake?

**Comment by barbudor** [comment]
ANALOG#AxDIV10 is meant to be used with an basic "ADC Input", not with "ADC Range"

So either 
- you don't use Range and you will have a value between 0 to 4095
- you use Range to be able to scale the value to 0..100 but you will need a more complex rule

The trigger would be ANALOG#Range1
And you can get similar use case here: https://tasmota.github.io/docs/Rules/#transmit-sensor-value-only-when-a-delta-is-reached
